Implement RoyalFlush

daelynj opened the pull request:
Return true if the hand is a royal flush. Return false if the hand is
not a royal flush. Don't bother checking the ranks of the hand if all
the suits of the cards aren't the same, thus return false early.

daelynj committed:
Implement RoyalFlush

Return true if the hand is a royal flush. Return false if the hand is
not a royal flush. Don't bother checking the ranks of the hand if all
the suits of the cards aren't the same, thus return false early.

jgayfer commented on lib/poker_hands/hand_rankings/royal_flush.rb:
```diff
@@ -1,5 +1,14 @@
+    royal_flush_ranks = ['A', 'K', 'Q', 'J', '10']
+    ranks_in_hand = []
+    
+    return false if hand.any? { |card| card.suit != hand.first.suit}
```
Are we passing in a `Hand` object or an array of `Card` objects?

daelynj replied:
In my code there is no such thing as a Hand object, in this case hand is an array of Card objects as you say. Does that need to be more clear, or do you ask because I had previously talked about implementing a Hand class (that I decided not to do).

jgayfer commented on lib/poker_hands/hand_rankings/royal_flush.rb:
```diff
@@ -1,5 +1,14 @@
+
+    hand.size.times do |card|
+      ranks_in_hand << hand[card].rank
+    end
```
Rubyism for you (assuming that `hand` is an array of `Card` objects)
```
hand.each do |card|
  ranks_in_hand << card.rank
end

# or

ranks_in_hand = hand.map(&:rank)
```

daelynj replied:
Right on

swcraig reviewed (commented):
A question to ask at this point is, could we leverage the (future) Flush logic here?

daelynj committed:
Ruby-ify finding the ranks in a hand

use a nicer looking Ruby-ism to map the rank value of cards in a hand
to an array.

daelynj reviewed (commented):
I've opted to comment out the code so I can better understand what is happening if I need to do the same thing in the future.

swcraig commented on lib/poker_hands/hand_rankings/royal_flush.rb:
```diff
@@ -5,9 +5,10 @@ def check(hand)
+    #  ranks_in_hand << card.rank
+    #end
 
     (royal_flush_ranks - ranks_in_hand).empty? ? true : false
```
We don't need to explicitly return the boolean here. The return value of a method in Ruby is that of the last evaluated expression.
